polkadot/xcm/src/v3/multilocation.rs

```diff
@@ -444,6 +444,20 @@ impl MultiLocation {
 			}
 		}
 	}
+
+	/// Return the MultiLocation subsection identifying the chain that `self` points to.
+	pub fn chain_location(mut self) -> MultiLocation {
```

Comment:
It kind of feels like it does some unnecessary assumptions about `MultiLocation` shape with `Junction::Parachain` and `Junction::GlobalConsensus` here in v3 xcm definition, maybe it is ok here.

I think maybe, this `chain_location` logic should/could also be a part of `trait AssetTransferSupport`  - maybe as function like `fn chain_location(location: &MultiLocation) -> MultiLocation`. With some default implementation inside or prepare some tuple based implementations structs like `PolkadotLikeParachainLocation` or `PolkadotLikeGlobalConsensusLocation`.  But also lets now over-engineered that :D

Reply:
I think it's fine for `v3` where it is actually defined, if `MultiLocation` shape changes in v4 or whatever, this function will change too for that version. Later we can also maybe add it to some trait (but not `AssetTransferSupport`), but like you say, I would not over-engineer right off the bat.